### DISCUSSION TITLE
How to run Python file on EdgeX, assuming my Python file has some logic to execute and send data to device

### DISCUSSION THREAD

**Discussion by samirlenka** [edgexfoundry/community#248] [Q&A]
I have a use case where I want to run a Python file on EdgeX and do some operations so that I could send data to Cloud on a regular interval.

**Comment by lenny-goodell** [comment]
Make your Python app look like a MQTT enabled Device and use Device MQTT communicate with it. i.e. Have the python app connect to MQTT Broken and listen to "commands" from Device MQTT. Enable auto events in the device definition in Device MQTT to make reads periodically or the Python app can internally send async data to Device MQTT.

See this How To guide which uses a Java Script in the same manner you would use Python.
https://docs.edgexfoundry.org/3.1/microservices/device/services/device-mqtt/Ch-ExamplesAddingMQTTDevice/
See here for more details about Device MQTT:
https://docs.edgexfoundry.org/3.1/microservices/device/services/device-mqtt/Purpose/
